dev/tasks/exec/tasks/exec.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const tools = require( '../../../lib/tools' );
+const git = require( '../../../lib/git' );
+const path = require( 'path' );
+const log = require( '../../../lib/log' );
+const merge = require( 'merge-stream' );
+
+/**
+ * @param {Function} execTask Task to use on each dependency.
+ * @param {String} ckeditor5Path Path to main CKEditor5 repository.
+ * @param {Object} packageJSON Parsed package.json file from CKEditor5 repository.
+ * @param {String} workspaceRoot Relative path to workspace root.
+ * @param {Boolean} dryRun
+ */
+module.exports = ( execTask, ckeditor5Path, packageJSON, workspaceRoot, params ) => {
+	const workspaceAbsolutePath = path.join( ckeditor5Path, workspaceRoot );
+
+	// Get all CKEditor dependencies from package.json.
+	const dependencies = tools.getCKEditorDependencies( packageJSON.dependencies );
+	const streams = merge();
+
+	if ( dependencies ) {
+		const directories = tools.getCKE5Directories( workspaceAbsolutePath );
+
+		if ( directories.length ) {
+			for ( let dependency in dependencies ) {
+				const repositoryURL = dependencies[ dependency ];
+				const urlInfo = git.parseRepositoryUrl( repositoryURL );
+				const repositoryAbsolutePath = path.join( ckeditor5Path, 'node_modules', dependency );
+
+				// Check if repository's directory already exists.
+				if ( directories.indexOf( urlInfo.name ) > -1 ) {
```

Comment:
It started as a copy of `ckeditor5/dev/tasks/dev/tasks/status.js` - there you can find most of the reasons behind this structure.